Update-HostsFile.ps1

```diff
@@ -0,0 +1,170 @@
+        $BlackListFilePath,
+        [switch]
+        $BackupOldHostsFile
+    )
+    if(-not ([Security.Principal.WindowsPrincipal][Security.Principal.WindowsIdentity]::GetCurrent()).IsInRole([Security.Principal.WindowsBuiltInRole]::Administrator))
+    {
+        throw 'This script requires local administrator privilege'
+    }
+
+    $BaseUri = 'https://raw.githubusercontent.com/StevenBlack/hosts/master'
+    if($PSBoundParameters.ContainsKey('All'))
+    {
+        $null = $PSBoundParameters.Remove('All')
+        $null = $PSBoundParameters.Add('Fakenews',$true)
+        $null = $PSBoundParameters.Add('Gambling',$true)
+        $null = $PSBoundParameters.Add('Social',$true)
+        $null = $PSBoundParameters.Add('Porn',$true)
+    }
+    
+    if($PSBoundParameters.Keys.Count -eq 0 -or $PSBoundParameters.ContainsKey('UnifiedOnly'))
+    {
+        Write-Verbose 'Processing UnifiedOnly Hosts file'
+        $downloadUri = $BaseUri + '/hosts'
+    }
+    elseif($PSBoundParameters.Keys.Count -ge 1 -and -not $PSBoundParameters.ContainsKey('All'))
+    {
+        
+        $downloadUri = $BaseUri + '/alternates/' + (( $PSBoundParameters.Keys|Where-Object {$_ -notmatch 'File'}|ForEach-Object {$_.ToLower()} | Sort-Object ) -join '-') + '/hosts'
+        Write-Verbose "Using $downloadUri"
+    }
+    try{
+    $webResponse = Invoke-WebRequest -Uri $downloadUri -UseBasicParsing 
+    }
+    catch
+    {
+      throw $_
+    }
+    
+    $hostsContent = $webResponse.Content
+    $hostsArray = New-Object System.Collections.ArrayList  
+    $null = $hostsArray.AddRange(($hostsContent -split '\n'))
+    if(-not [string]::IsNullOrEmpty($WhitelistFilePath) )
+    {
+        
+        $whiteListDomains = Get-Content -Path $WhitelistFilePath
+        $hostsToWhiteList = New-Object System.Collections.ArrayList
+        foreach($hostsEntry in $hostsArray)
+        {
+            if($whiteListDomains.Contains($hostsEntry.Split('\s')[1]))
+            {
+                $null = $hostsToWhiteList.Add($hostsEntry)
+            }
+        }
+        foreach($htwl in $hostsToWhiteList)
+        {
+            $null= $hostsArray.Remove($htwl)
+        }
+    }
+    if(-not [string]::IsNullOrEmpty($BlackListFilePath))
+    {
+        $null = $hostsArray.Add("")
+        $null =         $hostsArray.Add("# Custom Blacklistentries from $BlackListFilePath")
+        foreach($blacklistEntry in (Get-Content -Path $BlackListFilePath))
+        {
+            $null = $hostsArray.Add("0.0.0.0 $blacklistEntry")
+        }
+    }
+    if($BackupOldHostsFile)
+    {
+        Copy-Item (Join-Path $env:windir 'System32\drivers\etc\hosts') (Join-Path $env:windir 'System32\drivers\etc\hosts.bak')
+    }
+    try
+    {
+        $stream = [System.IO.StreamWriter]::new( (Join-Path $env:windir 'System32\drivers\etc\hosts') )
+        $hostsArray | ForEach-Object{ $stream.WriteLine( $_ ) }
+    }
+    finally
+    {
+        $stream.close()
+    }
+    $null = ipconfig.exe /flushdns

```

Comment:
@StevenBlack : Is this way of flushing the DNS cache still a sticky area for existing users of your hosts file?  Otherwise, this might be useful to add on the Python side too.

Reply:
@grantcarthew : Any input on this BTW?  Is this command (somewhat) backwards compatible for older Windows versions?

Reply:
Hi , yeah it should compatible with Win 7+ , the important point is the installed PowerShell version, it must be at least PowerShell v3